### PR TITLE
Gcr resource

### PR DESCRIPTION
**Pull request by slevenick**
Fixes: https://github.com/terraform-providers/terraform-provider-google/issues/4184

Add support for a GCR resource that ensures the backing bucket for a project/location combination exists. This allows for terraform to ensure the bucket exists prior to setting permissions on it. 

Normally the registry is created when the first image is pushed to it, but we cannot push images via Terraform. Performing the auth token handshake on the correct domain is enough to ensure that the bucket gets created by GCR.

Delete does not function on this resource so that we don't delete the bucket containing container images. This is not a function allowed via GCR, but can be done manually by deleting the bucket, so I decided to not include it.

This resource does a create or acquire on create, so import is not needed.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_container_registry`
```

### CODE REVIEW

File: third_party/terraform/resources/resource_container_registry.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"self_link": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"url": {
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_container_registry.go]: Not used anywhere / documented

File: third_party/terraform/resources/resource_container_registry.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+func resourceContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceContainerRegistryCreate,
+		Read:   resourceContainerRegistryRead,
+		Delete: resourceContainerRegistryDelete,
+
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:     schema.TypeString,
+				Default:  "US",
```
Comment (rileykarson) [inline third_party/terraform/resources/resource_container_registry.go]:
```suggestion
```

Can we omit this so that the "global" location is the default?

Comment (slevenick) [reply third_party/terraform/resources/resource_container_registry.go]: Yeah, the default location is also `US` but without a prefix
`gcr.io hosts images in data centers in the United States, but the location may change in the future`

I'll change it

**Comment by thomashutcheson-msm** [comment]
@slevenick I am sorry to gate crash this thread, are you able to glance an eye over this other issue, https://github.com/kreuzwerker/terraform-provider-docker/issues/189 

I have followed the thread from https://github.com/hashicorp/terraform-provider-google/issues/4184
which lead me here and can see you have had experience looking into building and pushing a docker image via terraform, which is what I am in need of.

**Comment by slevenick** [comment]
> @slevenick I am sorry to gate crash this thread, are you able to glance an eye over this other issue, [kreuzwerker/terraform-provider-docker#189](https://github.com/kreuzwerker/terraform-provider-docker/issues/189)
> 
> I have followed the thread from [hashicorp/terraform-provider-google#4184](https://github.com/hashicorp/terraform-provider-google/issues/4184)
> which lead me here and can see you have had experience looking into building and pushing a docker image via terraform, which is what I am in need of.

Nope, I haven't tried pushing a docker image via TF.

**Pull request merged**